src/graph.py

```diff
@@ -540,68 +550,70 @@ def randomizations(self, noisy_edges_reps, random_terminals_reps):
                 #######          Grid Search          #######
     ###########################################################################
 
-    def _eval_pcsf(self, params):
+    def _eval_PCSF_runs(self, params):
         """
-        Convenience methods which sets parameters and performs PCSF
-        """
-        self._reset_hyperparameters(params=params)
-        paramstring = 'G_'+str(self.params.g)+'_B_'+str(self.params.b)+'_W_'+str(self.params.w)
-        print(paramstring)
-        return (paramstring, self.pcsf())
+        Convenience method which sets parameters and performs PCSF randomizations.
 
+        Arguments: 
+            params (dict): params with which to run the program
 
-    def _grid_pcsf(self, prize_file, Gs, Bs, Ws):
+        Returns: 
+            str: Parameter values in string format 
+            networkx.Graph: forest
+            networkx.Graph: augmented_forest
         """
-        Internal function which executes pcsf at every point in a parameter grid.
-        Subroutine of `grid_search`.
-
-        Arguments:
-            prize_file (str): filepath
-            Gs (list): Values of gamma
-            Bs (list): Values of beta
-            Ws (list): Values of omega
 
-        Returns:
-            list: list of tuples of vertex indices and edge indices
-        """
+        self._reset_hyperparameters(params=params)
+        paramstring = "W_{:04.2f}_B_{:04.2f}_G_{:d}".format(self.params.w, self.params.b, self.params.g)
 
-        self.prepare_prizes(prize_file)
-        parameter_permutations = [{'g':g,'b':b,'w':w} for (g, b, w) in product(Gs, Bs, Ws)]
-        results = list(map(self._eval_pcsf, parameter_permutations))
+        if params["noisy_edge_reps"] + params["random_terminals_reps"] == 0:
+            logger.info("Single PCSF runs for " + paramstring)
+        else: 
+            logger.info("Randomizations for " + paramstring)
 
-        return results
+        forest, augmented_forest = self.randomizations(params["noisy_edge_reps"], params["random_terminals_reps"])
+        
+        return paramstring, forest, augmented_forest
 
 
-    def _grid_pcsf_parallel(self, prize_file, Gs, Bs, Ws):
+    def grid_randomization(self, prize_file, Ws, Bs, Gs, noisy_edges_reps, random_terminals_reps):
         """
-        Internal function which executes pcsf at every point in a parameter grid.
-        Subroutine of `grid_search`. This version runs using python multiprocessing
+        Macro function which performs grid search or randomizations or both. 
 
         Arguments:
             prize_file (str): filepath
             Gs (list): Values of gamma
             Bs (list): Values of beta
             Ws (list): Values of omega
+            noisy_edges_reps (int): Number of robustness experiments
+            random_terminals_reps (int): Number of specificity experiments
 
         Returns:
-            list: list of tuples of vertex indices and edge indices
+            results (dict)
```

Comment:
Should follow the `return type: explanation of return value` form

Reply:
fe2e6617c10145e20029f982232da268a976dea5